server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -171,29 +183,50 @@ protected void parseConfiguration(ServerConfiguration conf) {
     poolNameToRegexPattern = new HashMap<>();
     for (Entry<String,String> table : t.tableIdMap().entrySet()) {
       tableIdToTableName.put(table.getValue(), table.getKey());
+      conf.getTableConfiguration(table.getValue()).addObserver(this);
       Map<String,String> customProps = conf.getTableConfiguration(table.getValue()).getAllPropertiesWithPrefix(Property.TABLE_ARBITRARY_PROP_PREFIX);
       if (null != customProps && customProps.size() > 0) {
         for (Entry<String,String> customProp : customProps.entrySet()) {
           if (customProp.getKey().startsWith(HOST_BALANCER_PREFIX)) {
+            if (customProp.getKey().equals(HOST_BALANCER_OOB_CHECK_KEY) || customProp.getKey().equals(HOST_BALANCER_POOL_RECHECK_KEY)
+                || customProp.getKey().equals(HOST_BALANCER_REGEX_USING_IPS_KEY) || customProp.getKey().equals(HOST_BALANCER_REGEX_MAX_MIGRATIONS_KEY)) {
+              continue;
+            }
             String tableName = customProp.getKey().substring(HOST_BALANCER_PREFIX.length());
             String regex = customProp.getValue();
             poolNameToRegexPattern.put(tableName, Pattern.compile(regex));
           }
         }
       }
     }
-    String oobProperty = conf.getConfiguration().get(HOST_BALANCER_OOB_CHECK);
+    String oobProperty = conf.getConfiguration().get(HOST_BALANCER_OOB_CHECK_KEY);
     if (null != oobProperty) {
       oobCheckMillis = AccumuloConfiguration.getTimeInMillis(oobProperty);
     }
     String poolRecheckProperty = conf.getConfiguration().get(HOST_BALANCER_POOL_RECHECK_KEY);
     if (null != poolRecheckProperty) {
       poolRecheckMillis = AccumuloConfiguration.getTimeInMillis(poolRecheckProperty);
     }
-    String ipBased = conf.getConfiguration().get(HOST_BALANCER_REGEX_USING_IPS);
+    String ipBased = conf.getConfiguration().get(HOST_BALANCER_REGEX_USING_IPS_KEY);
     if (null != ipBased) {
       isIpBasedRegex = Boolean.parseBoolean(ipBased);
     }
+    String migrations = conf.getConfiguration().get(HOST_BALANCER_REGEX_MAX_MIGRATIONS_KEY);
+    if (null != migrations) {
+      maxTServerMigrations = Integer.parseInt(migrations);
+    }
+    LOG.info("{}", this);
```

Comment:
good info.....but instead of relying on toString use something like this.dumpConfiguration